### PR TITLE
Implementation with jersey

### PR DESCRIPTION
**Pull request by kdrtibor**

### CODE REVIEW

File: src/main/java/com/example/rest/UserStoryResource.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    @Produces(MediaType.APPLICATION_JSON)
+    public String updateUserStory(@PathParam("userStoryId") int userStoryId,UserStory newStory) {
+        for (UserStory us : UserStoryRepository.getAll()) {
+            if (us.getId() == userStoryId) {
+                UserStoryRepository.remove(us);
+                UserStoryRepository.add(newStory);
+                return "Updated successfully";
+            }
+        }
+        return "Object to update not found";
```
Comment (andrast0th) [inline src/main/java/com/example/rest/UserStoryResource.java]: By convention, if an object is not found, the server should send 404, if a rest update is succesfull, it should send the updated object's json

File: src/main/java/com/example/rest/UserStoryResource.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    @Path("/{userStoryId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String deleteUserStory(@PathParam("userStoryId") int userStoryId){
+        for (UserStory us : UserStoryRepository.getAll()) {
+            if (us.getId() == userStoryId) {
+                UserStoryRepository.remove(us);
+                return "Deleted successfully";
+            }
+        }
+        return "Object to delete not found";
```
Comment (andrast0th) [inline src/main/java/com/example/rest/UserStoryResource.java]: 404

File: src/main/java/com/example/rest/UserStoryResource.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return null;
+    }
+
+    @POST
+    @Secured
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public String postUserStory(UserStory story){
+        UserStoryRepository.add(story);
+        return "successfully posted";
```
Comment (andrast0th) [inline src/main/java/com/example/rest/UserStoryResource.java]: return the created object json

File: pom.xml [hunk 4]
```diff
@@ -1,25 +1,91 @@
...
+        <dependency>
+            <groupId>org.eclipse.jetty</groupId>
+            <artifactId>jetty-servlet</artifactId>
+            <version>9.2.11.v20150529</version>
+        </dependency>
+        <dependency>
+            <groupId>log4j</groupId>
+            <artifactId>log4j</artifactId>
+            <version>1.2.12</version>
+        </dependency>
```
Comment (andrast0th) [inline pom.xml]: I see the logger added here. But where is it used?

File: src/main/java/com/example/rest/UserStoryResource.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @PUT
+    @Secured
+    @Path("/{userStoryId}")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String updateUserStory(@PathParam("userStoryId") int userStoryId,UserStory newStory) {
```
Comment (andrast0th) [inline src/main/java/com/example/rest/UserStoryResource.java]: formatting: space after comma

**Review by andrast0th — CHANGES_REQUESTED** [review]

**Comment by andrast0th** [comment]
The commits regarding authentication seem to be missing? Were they pushed?

File: pom.xml (hunk 4 above)
Comment (kdrtibor) [reply pom.xml]: i think i added that by mistake when i tried to resolve an error in the compiler, followed a suggestion of intellij which said download maven dependency or something similar. Later i managed to fix the error but forgot about the dependency.

**Review by andrast0th — COMMENTED** [review]
Cool!

File: pom.xml (hunk 4 above)
Comment (andrast0th) [reply pom.xml]: that's alright, logging is cool, we can make a story to add logger support for the project, we'll see

File: src/main/java/com/example/rest/UserStoryResource.java (hunk 5 above)
Comment (andrast0th) [reply src/main/java/com/example/rest/UserStoryResource.java]: I will test this out :D

File: src/main/java/com/example/App.java [hunk 6]
```diff
@@ -20,7 +20,7 @@ public static void main(String[] args)
         // Tells the Jersey Servlet which REST service/class to load.
         jerseyServlet.setInitParameter(
                 "jersey.config.server.provider.packages",
-                "com.example.rest");
```
Comment (andrast0th) [inline src/main/java/com/example/App.java]: Why was this changed?

File: src/main/java/com/example/App.java [hunk 7]
```diff
@@ -1,10 +1,33 @@
...
+        jettyServer.setHandler(context);
+
+        ServletHolder jerseyServlet = context.addServlet(
+                org.glassfish.jersey.servlet.ServletContainer.class, "/api/*");
+        jerseyServlet.setInitOrder(0);
+
+        // Tells the Jersey Servlet which REST service/class to load.
+        jerseyServlet.setInitParameter(
+                "jersey.config.server.provider.packages",
+                "com.example");
```
Comment (andrast0th) [inline src/main/java/com/example/App.java:23]: I would move the authentication endpoint and filter into rest package, as they are part of the api.

**Review by andrast0th — CHANGES_REQUESTED** [review]

**Review by andrast0th — APPROVED** [review]
Noice!
